core/scripts/common/polygonedge.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/umbracle/fastrlp"
+)
+
+type Nonce [8]byte
+
+var (
+	nonceT = reflect.TypeOf(Nonce{})
+)
+
+func (n Nonce) String() string {
+	return EncodeToHex(n[:])
+}
+
+// MarshalText implements encoding.TextMarshaler
+func (n Nonce) MarshalText() ([]byte, error) {
+	return []byte(n.String()), nil
+}
+
+// UnmarshalJSON parses a nonce in hex syntax.
+func (n *Nonce) UnmarshalJSON(input []byte) error {
+	return hexutil.UnmarshalFixedJSON(nonceT, input, n[:])
+}
+
+type ExtraData string
+
+func (e ExtraData) Decode() ([]byte, error) {
+	return DecodeHex(string(e))
+}
+
+// Header represents a block header in the Ethereum blockchain.
+type PolygonEdgeHeader struct {
+	ParentHash   common.Hash    `json:"parentHash"       gencodec:"required"`
+	Sha3Uncles   common.Hash    `json:"sha3Uncles"       gencodec:"required"`
+	Miner        common.Address `json:"miner"`
+	StateRoot    common.Hash    `json:"stateRoot"        gencodec:"required"`
+	TxRoot       common.Hash    `json:"transactionsRoot" gencodec:"required"`
+	ReceiptsRoot common.Hash    `json:"receiptsRoot"     gencodec:"required"`
+	LogsBloom    types.Bloom    `json:"logsBloom"        gencodec:"required"`
+	Difficulty   hexutil.Uint64 `json:"difficulty"       gencodec:"required"`
+	Number       hexutil.Uint64 `json:"number"           gencodec:"required"`
+	GasLimit     hexutil.Uint64 `json:"gasLimit"         gencodec:"required"`
+	GasUsed      hexutil.Uint64 `json:"gasUsed"          gencodec:"required"`
+	Timestamp    hexutil.Uint64 `json:"timestamp"        gencodec:"required"`
+	ExtraData    ExtraData      `json:"extraData"        gencodec:"required"`
+	MixHash      common.Hash    `json:"mixHash"`
+	Nonce        Nonce          `json:"nonce"`
+	Hash         common.Hash    `json:"hash"`
+
+	// BaseFee was added by EIP-1559 and is ignored in legacy headers.
+	BaseFee    hexutil.Uint64 `json:"baseFeePerGas"`
+	BaseFeeAlt hexutil.Uint64 `json:"baseFee,omitempty"`
+}
+
+func GetPolygonEdgeRLPHeader(polygonEdgeClient *rpc.Client, blockNum *big.Int) (rlpHeader []byte, hash string, err error) {
+	var h PolygonEdgeHeader
+	err = polygonEdgeClient.Call(&h, "eth_getBlockByNumber", "0x"+blockNum.Text(16), true)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to get poloygon-edge header: %+v", err)
+	}
+
+	ar := &fastrlp.Arena{}
+	val, err := MarshalRLPWith(ar, &h)
+	if err != nil {
+		return nil, "", err
+	}
+	// fastrlp.DefaultArenaPool.Put(ar)
+
+	dst := make([]byte, 0)
+	dst = val.MarshalTo(dst)
+
+	return dst, h.Hash.String(), err
+}
+
+// MarshalRLPWith marshals the header to RLP with a specific fastrlp.Arena
+// Adding polygon-edge as a dependency caused a lot of issues with conflicting
+// dependency version with other libraries in this repo and some methods being
+// referenced from older versions
+// Reference: https://github.com/0xPolygon/polygon-edge/blob/develop/types/rlp_marshal.go#L73C50-L73C53
```

Comment:
I wonder if Polygon can change this implementation in the future. How bad were the dependency issues?

Reply:
The dependency issues seemed like a rabbit hole. After fixing the dependency version issues, I ran into some undefined methods problem, probably due to methods being referenced from older versions. 

We should consider relying on polygon-edge in future, when the dependency issues resolve.